### ISSUE TITLE
Delta as pager with side-by-side is no longer rendering full width

### ISSUE THREAD

**Issue by simonvizzini** [jesseduffield/lazygit#1582]
**Describe the bug**
I _think_ this issue was introduced with the latest lazygit version, or maybe it is an issue with Delta itself, I'm not really sure, let me know what you think.

I have configured Delta as pager according to the documentation but now the diff is not taking up the full width as can be seen in the screenshot below.

This only happens when using Delta's "side-by-side" feature.

**To Reproduce**
1. Setup Delta as pager in lazygits config.yml:

````
gui:
  theme:
    selectedLineBgColor:
      - reverse
    selectedRangeBgColor:
      - reverse
git:
  paging:
    colorArg: always
    pager: delta -s --dark --paging=never
````

2. Check diff within lazygit and notice that its not full width anymore, see screenshot.

**Expected behavior**
Diff output should take up all the available width, like before.

**Screenshots**
Delta side by side:
![image](https://user-images.githubusercontent.com/10862380/142009005-d4bddfa0-6258-4d57-b319-f75beed80ef2.png)
Delta default:
![image](https://user-images.githubusercontent.com/10862380/142009686-7ebdecd7-c002-4129-9cf0-5e482d3bb12e.png)

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Lazygit Version v0.31.3
 - Delta Version 0.9.2

**Comment by blm** [comment]
I did some stuff with git bisect and I think I found git commit 2fc1498517523a20a3080816ec50ee9e7fbe533d was the last commit with full width diff and git commit 802cfb1a0436568c72fc998249f10f8150b352a3 was the first one with reduced width diff display.

Command history:

     git checkout 802cfb1a0436568c72fc998249f10f8150b352a3
     go build
     ./lazygit
     #screenshot
     git checkout 2fc1498517523a20a3080816ec50ee9e7fbe533d
     go build
     ./lazygit
      #screenshot

See attached screenshots.  Will this help?

Screenshot of 802cfb1a0436568c72fc998249f10f8150b352a3
![screenshot first bad 802cfb1a0436568c72fc998249f10f8150b352a3](https://user-images.githubusercontent.com/7706/142160366-da63180d-4e4d-425a-a558-fb534d364739.png)

Screenshot of 2fc1498517523a20a3080816ec50ee9e7fbe533d

![screenshot last good 2fc1498517523a20a3080816ec50ee9e7fbe533d](https://user-images.githubusercontent.com/7706/142160682-3247782a-a103-4bc5-941f-877b19e7fa37.png)

**Comment by nikbrunner** [comment]
I have the same issue with the same settings

**Comment by andrelouw** [comment]
Same for me.

**Comment by blm** [comment]
I have been playing around with this issue.  I noticed that the squares at the end of the line in the above screenshots are characters delta puts in to tell people the line was wrapped.  delta uses what it thinks as the size of the screen to decide where to wrap lines.  When you count characters delta shows up to and including the line wrap character you find it is 80 line the width of a vt100 display.

Then I replaced the pager with some rust code that I found on the Internet that I think uses the same method to determine the terminal size
```rust
extern crate console;

use std::io;
use std::thread;
use std::time::Duration;

use console::{style, Term};

fn write_chars() -> io::Result<()> {
    let term = Term::stdout();
    let (heigth, width) = term.size();
    term.write_line(&heigth.to_string())?;
    term.write_line(&width.to_string())?;
    Ok(())
}

fn main() {
    write_chars().unwrap();
}
```
This code thinks the terminal is 24x80

Screenshot:

![dummy rust program as pager](https://user-images.githubusercontent.com/7706/142831019-a962c414-546d-4c90-9f3b-1010028542f9.png)

Then I found some other rust code.  I think the console crate uses this module to determine the terminal size.  I also put it in a loop to see if resizes work.

```rust
use std::{thread, time};


#[cfg(windows)]
fn run() {
    use std::os::windows::io::RawHandle;
    use winapi::um::processenv::GetStdHandle;
    use winapi::um::winbase::{STD_ERROR_HANDLE, STD_INPUT_HANDLE, STD_OUTPUT_HANDLE};

    let stdout = unsafe { GetStdHandle(STD_OUTPUT_HANDLE) } as RawHandle;
    println!(
        "Size from terminal_size_using_handle(stdout): {:?}",
        terminal_size::terminal_size_using_handle(stdout)
    );

    let stderr = unsafe { GetStdHandle(STD_ERROR_HANDLE) } as RawHandle;
    println!(
        "Size from terminal_size_using_handle(stderr): {:?}",
        terminal_size::terminal_size_using_handle(stderr)
    );

    let stdin = unsafe { GetStdHandle(STD_INPUT_HANDLE) } as RawHandle;
    println!(
        "Size from terminal_size_using_handle(stdin):  {:?}",
        terminal_size::terminal_size_using_handle(stdin)
    );
}

#[cfg(not(windows))]
fn run() {
    println!(
        "Size from terminal_size_using_fd(stdout):     {:?}",
        terminal_size::terminal_size_using_fd(libc::STDOUT_FILENO)
    );
    println!(
        "Size from terminal_size_using_fd(stderr):     {:?}",
        terminal_size::terminal_size_using_fd(libc::STDERR_FILENO)
    );
    println!(
        "Size from terminal_size_using_fd(stdin):      {:?}",
        terminal_size::terminal_size_using_fd(libc::STDIN_FILENO)
    );
}

fn main() {
    let delay = time::Duration::from_secs(3);
    loop {
        println!(
            "Size from terminal_size():                    {:?}",
            terminal_size::terminal_size()
        );

        run();
        println!("sleeping for 3  sec ");
        thread::sleep(delay);
    }
}

```
This code says the terminal is 0 x 0 until you resize it.  But the size seems to go back to 0 x 0 .

I also tried some C code from http://www.rkoucha.fr/tech_corner/sigwinch.html

The size of the terminal reported by this code is random.

-----

One thing I did notice was there is the start function

```go
func Start(cmd *exec.Cmd) (*os.File, error)
```

But there is also StartWithSize

```go
func StartWithSize(cmd *exec.Cmd, ws *Winsize) (*os.File, error)
```
which might be better for running commands like what we do here (git diff with delta as the pager).  This way the terminal size is definitely there before the command starts.

**Comment by jesseduffield** [comment]
Thanks everyone for the great troubleshooting! I've got a PR up now that resolves the issue https://github.com/jesseduffield/lazygit/pull/1592

**Comment by jesseduffield** [comment]
fixed in v0.31.4 (releasing now)
